core/src/Module/Sql/DataSource/RawSqlDataSource.php

```diff
@@ -81,31 +97,65 @@ public function getAll(Query $query): array {
 
   /**
    * @psalm-suppress LessSpecificImplementedReturnType
+   *
    * @param Query      $query
    * @param mixed|null $entity
    *
    * @return mixed
-   * @throws QueryNotSupportedException
    */
   public function put(Query $query, mixed $entity = null): mixed {
```

Comment:
@orioljp please, ping me to talk about this changes, the new function transaction and how we ware using it on executePutQuery.